packages/react/src/drafts/UnderlinePanels/UnderlinePanels.features.stories.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react'
+import type {Meta} from '@storybook/react'
+import {INITIAL_VIEWPORTS} from '@storybook/addon-viewport'
+import UnderlinePanels from './UnderlinePanels'
+import type {ComponentProps} from '../../utils/types'
+import {
+  CodeIcon,
+  CommentDiscussionIcon,
+  EyeIcon,
+  GearIcon,
+  GitPullRequestIcon,
+  GraphIcon,
+  PlayIcon,
+  ProjectIcon,
+  ShieldLockIcon,
+} from '@primer/octicons-react'
+
+export default {
+  title: 'Drafts/Components/UnderlinePanels/Features',
+  component: UnderlinePanels,
+} as Meta<ComponentProps<typeof UnderlinePanels>>
+
+export const SelectedTab = () => (
+  <UnderlinePanels aria-label="Select a tab" id="tab-panels">
+    <UnderlinePanels.Tab>Tab 1</UnderlinePanels.Tab>
+    <UnderlinePanels.Tab aria-selected={true}>Tab 2</UnderlinePanels.Tab>
+    <UnderlinePanels.Tab>Tab 3</UnderlinePanels.Tab>
+    <UnderlinePanels.Panel>Panel 1</UnderlinePanels.Panel>
+    <UnderlinePanels.Panel>Panel 2</UnderlinePanels.Panel>
+    <UnderlinePanels.Panel>Panel 3</UnderlinePanels.Panel>
+  </UnderlinePanels>
+)
+
+export const LabelledByExternalElement = () => (
+  <>
+    <h2 id="my-heading">UnderlinePanels example</h2>
+    <UnderlinePanels aria-labelledby="my-heading">
+      <UnderlinePanels.Tab>Tab 1</UnderlinePanels.Tab>
+      <UnderlinePanels.Tab>Tab 2</UnderlinePanels.Tab>
+      <UnderlinePanels.Tab>Tab 3</UnderlinePanels.Tab>
+      <UnderlinePanels.Panel>Panel 1</UnderlinePanels.Panel>
+      <UnderlinePanels.Panel>Panel 2</UnderlinePanels.Panel>
+      <UnderlinePanels.Panel>Panel 3</UnderlinePanels.Panel>
+    </UnderlinePanels>
+  </>
+)
+
+export const WithIcons = () => (
+  <UnderlinePanels aria-label="Tabs with icons">
+    <UnderlinePanels.Tab icon={CodeIcon}>Code</UnderlinePanels.Tab>
+    <UnderlinePanels.Tab icon={EyeIcon}>Issues</UnderlinePanels.Tab>
+    <UnderlinePanels.Tab icon={GitPullRequestIcon}>Pull requests</UnderlinePanels.Tab>
+    <UnderlinePanels.Panel>Code panel</UnderlinePanels.Panel>
+    <UnderlinePanels.Panel>Issues panel</UnderlinePanels.Panel>
+    <UnderlinePanels.Panel>Pull requests panel</UnderlinePanels.Panel>
+  </UnderlinePanels>
+)
+
+export const WithIconsHiddenOnNarrowScreen = () => (
+  <UnderlinePanels aria-label="Tabs with icons">
```

Comment:
I experience flickering in this story and ot might not be specific to only this story though but not sure why it happens. When I resize a scroll bar (I think?) appears for a hot second and then drops the icons. Sharing a screen recording, let me know what you think

_Video preview describes the behaviour I mentioned just above_

https://github.com/primer/react/assets/1446503/92fe7758-5801-4d0b-99ca-4fcf2968f212

Reply:
I'm not sure how we'd get around this _and_ keep the tablist scrollable. This can be something we tackle when/if we introduce the same overflow behavior we have for UnderilneNav.

I wouldn't consider this a blocker, but would you or other reviewers?

Reply:
I wouldn't say a blocker for this PR either but I think we should resolve it eventually. Could we set the overflow to hidden by default and only set to auto when `data-icons-visible="false"` or wise versa  or something along those lines?